src/pyodbc.pyi

```diff
@@ -404,11 +442,11 @@ def getDecimalSeparator() -> str: ...
 
 # https://www.python.org/dev/peps/pep-0249/#connect
 def connect(connstring: str,
-            /, *,  # positional-only parameters before, named-only after
+            /, *,  # only positional parameters before, only named parameters after
             autocommit: bool = False,
             encoding: str = 'utf-16le',
             ansi: bool = False,
             readonly: bool = False,
-            timeout: int,
-            attrs_before: dict,
+            timeout: int = 0,
+            attrs_before: dict = {},
```

Comment:
Hi. I have little experience in pyodbc, but typically things like `{}` or `[]` are not good for default values. Are you sure it's correct?

Reply:
I don't know what you mean by 'not good for default values", but the default is empty so it is correct..

Reply:
See https://github.com/satwikkansal/wtfpython#-beware-of-default-mutable-arguments

Reply:
You're certainly correct that real code should not typically have mutable values as default parameter values (because default values are created only once, not every time the function is called).  This code however is in the .pyi file which is used only for typing.  If these two defaults are left out of the function definition, it looks like the two parameters are mandatory which is not the case.

Reply:
Also, these functions are implemented in native code.